zulipterminal/ui_tools/messages.py

```diff
@@ -152,9 +153,15 @@ def stream_header(self) -> Any:
         assert self.stream_id is not None
         color = self.model.stream_dict[self.stream_id]["color"]
         bar_color = f"s{color}"
+        stream_access_type = self.model.stream_access_type(self.stream_id)
+        try:
+            stream_icon = STREAM_ACCESS_TYPE[stream_access_type]["icon"]
+        except KeyError:
+            stream_icon = INVALID_MARKER
```

Comment:
Is this just for the tests?

The model method should only ever return a valid index into STREAM_ACCESS_TYPE, so we shouldn't need this try/except otherwise.

Reply:
Yes, I've added the try/except in order to clear tests. Every time I ran the tests it gave a key error on the line with STREAM_ACCESS_TYPE. The error message I kept getting was :
`KeyError: <MagicMock name='mock.stream_access_type()' id='4817710544'>`
 I suspect that the test is missing a parameter to access the STREAM_ACCESS_TYPE, that's why the test results in a key error. I tried to look over the test to figure it out but wasn't able to. 
Do you have any ideas?

Reply:
You have the right idea as to the fundamental issue here - since the mock is very general, it doesn't have specific values and so doesn't correctly key into the dict. The simplest way to do this is likely to set `stream_access_type.return_value="public"` in a convenient location, at least to make the tests run and enable you to adjust the test cases.

It'd be useful to have tests with other icons, but that's a different change.

Reply:
That seems to be working, but I have to fiddle with it more to see where exactly to put it so it works. I'm currently putting it in several test in TestMessageBox class.

Reply:
The simplest approach may be to insert it into the two fixtures which are used in all/most of the tests.

Reply:
I have looked over conftest.py came to the conclusion that msg_box is the fixture to use for this job. But I'm having trouble getting it to work as I can't get it to return "public" when it's called. Do you have any suggestions?

Reply:
If you use something like I mentioned above it should work. I'd suggest you push another commit with the test changes, as that way people can see what you have and how you might fix it.